Chrismarcel/iReporter
# #163412335 Implement edit report functionality

__What does this PR do?__
This PR implements the edit report functionality by allowing an authenticated user to edit reports they have created. The updates they make on the front end is persisted in the database.

__Description of tasks__
- Ensure that authenticated users can edit their reports
- Ensure that changes made the reports on the front end gets persisted to the database

__How to test this PR__
- Visit https://chrismarcel.github.io/iReporter/UI/login.html
- Login
- Visit the given link https://chrismarcel.github.io/iReporter/UI/edit-report.html?type={type}&id={id} where type is the incident type which can be either red-flags or interventions and the id is the id of the post you created eg https://chrismarcel.github.io/iReporter/UI/edit-report.html?type=red-flags&id=1 
- You should be presented with a form that would enable you to update your report.
- Hit the submit button
- Click on the 'View Records' link on the navigation menu
- You should be redirected to a page where you should see your updated posts

__Link to relevant stories__
[#163412335](https://www.pivotaltracker.com/story/show/163412335)
